src/azureStorageExplorer/AzureStorageFS.ts

```
@@ -329,7 +377,66 @@ export class AzureStorageFS implements vscode.FileSystemProvider {
         }
     }
 
+    private getRootName(uri: vscode.Uri): string {
+        const match: RegExpMatchArray | null = uri.path.match(/^\/[^\/]*\/?/);
+        return match ? match[0] : '';
+    }
+
+    private verifyUri(uri: vscode.Uri, rootName?: string): vscode.Uri {
+        // Remove slashes from rootName for consistency
+        rootName = (rootName || this.getRootName(uri)).replace(/\//g, '');
+
+        if (rootName in this._queryCache) {
+            if (this._queryCache[rootName].invalid) {
+                throw new Error('Invalid URI query cache.');
+            }
+
+            if (uri.query) {
+                if (this._queryCache[rootName].query !== uri.query) {
+                    this._queryCache[rootName].invalid = true;
+                    throw new Error('Invalid URI query cache.');
```

I don't think this should throw an error. The current operation will work fine since the uri already has a query. The cache only matters for operations where we _don't_ have a query